Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm

```diff
@@ -208,113 +370,24 @@ fragment float4 fragmentShader(RasterizerData in [[stage_in]],
-                    bestDevice = device;
-                    bestFormat = format;
-                    bestDimDiff = dimDiff;
-
-                    // Check if we got an exact match, and exit the loop early in this case.
-                    if (static_cast<uint32_t>(dimensions.width) == maxWidth && static_cast<uint32_t>(dimensions.height) == maxHeight)
-                    {
-                        foundExactMatch = true;
-                        break;
-                    }
-                }
-            }
-
-            if (foundExactMatch)
-            {
-                break;
-            }
-        }
-
-        // If no matching device, throw an error with the message "ConstraintError" which matches the behavior in the browser.
-        if (bestDevice == nullptr)
-        {
-            return arcana::task_from_error<CameraDimensions>(std::make_exception_ptr(std::runtime_error{"ConstraintError: Unable to match constraints to a supported camera configuration."}));
-        }
-
         // Lock camera device and set up camera format. If there a problem initialising the camera it will give an error.
-        [bestDevice lockForConfiguration:&error];
+        [cameraDevice->implData->avDevice lockForConfiguration:&error];
         if (error != nil)
         {
             return arcana::task_from_error<CameraDimensions>(std::make_exception_ptr(std::runtime_error{"Failed to lock camera"}));
         }
 
-        [bestDevice setActiveFormat:bestFormat];
-        AVCaptureDeviceInput *input{[AVCaptureDeviceInput deviceInputWithDevice:bestDevice error:&error]};
-        [bestDevice unlockForConfiguration];
+        [cameraDevice->implData->avDevice setActiveFormat:resolution->implData->avDeviceFormat];
+        AVCaptureDeviceInput *input{[AVCaptureDeviceInput deviceInputWithDevice:cameraDevice->implData->avDevice error:&error]};
+        [cameraDevice->implData->avDevice unlockForConfiguration];
 
         // Capture the format dimensions.
-        CMVideoFormatDescriptionRef videoFormatRef{static_cast<CMVideoFormatDescriptionRef>(bestFormat.formatDescription)};
+        CMVideoFormatDescriptionRef videoFormatRef{static_cast<CMVideoFormatDescriptionRef>(resolution->implData->avDeviceFormat.formatDescription)};
         CMVideoDimensions dimensions{CMVideoFormatDescriptionGetDimensions(videoFormatRef)};
 #else
-        UNUSED(maxWidth);
-        UNUSED(maxHeight);
-        UNUSED(frontCamera);
         AVCaptureDevice* captureDevice{[AVCaptureDevice defaultDeviceWithMediaType:AVMediaTypeVideo]};
         AVCaptureDeviceInput *input{[AVCaptureDeviceInput deviceInputWithDevice:captureDevice error:&error]};
-        CMVideoFormatDescriptionRef videoFormatRef{static_cast<CMVideoFormatDescriptionRef>(captureDevice.activeFormat.formatDescription)};
+        CMVideoFormatDescriptionRef videoFormatRef{static_cast<CMVideoFormatDescriptionRef>(resolution->implData->avDeviceFormat.formatDescription)};
```

Comment:
Is iOS vs Mac OS branch still needed? It seems like maybe we can have these work the same on iOS and Mac OS since the device selection has been moved up to GetCameraDevices.

Reply:
I got back around to testing this and you are indeed correct. I was able to remove the main chunk of differing code between iOS and MacOS. There are still a few sporadic chunks that only work on iOS due to the camera API's being more robust on iOS. For the most part though they are now aligned and use a single implementation.